internal/backend/fs.go

```diff
@@ -163,6 +166,37 @@ func (s *storageFS) GetObjectWithGeneration(bucketName, objectName string, gener
 	return Object{}, errors.New("not implemented: fs storage type does not support versioning yet")
 }
 
+var crc32cTable = crc32.MakeTable(crc32.Castagnoli)
+
+func crc32cChecksum(content []byte) []byte {
+	checksummer := crc32.New(crc32cTable)
+	checksummer.Write(content)
+	return checksummer.Sum(make([]byte, 0, 4))
+}
+
+func encodedChecksum(checksum []byte) string {
```

Comment:
Since this and `encodedHash` are the same, could we unify them (with a common name)? Or maybe inline the call to `base64.StdEncoding.EncodeToString`?

Reply:
Can do, like I said it's a copy and paste job. Ideally, the md5 and crc code should be extracted and used by `fs.go` and `upload.go` but my golang knowledge is poor!